Possible bug with r-basics-with-tabular-data.md

drjwbaker opened the pull request:
I think this is wrong. For me l60..

> Take a look at the `AirPassengers` data set by typing `data(AirPassengers)` into the console

..doesn't work. What does work is just typing `AirPassengers`. Can you check this please @anisa-hawes before merging? (as it may just be my setup!)

### Checklist

- [x] Assign yourself in the "Assignees" menu
- [x] Assign at least one individual or team to "Reviewers"
  - [x] if the text needs to be translated, assign the relevant language team(s) as "Reviewers" and tag both the team as well as the managing edtor in your PR. Please follow the [translation request guidelines](https://github.com/programminghistorian/jekyll/wiki/Requesting-Translation-Guidelines) when writing your PR description
- [x] Add the appropriate "Label"
- [x] [Ensure the status checks pass](https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions#4-check-your-pr-status)
- [x] [Check the live preview of your PR on Netlify](https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions#5-preview-how-your-pr-looks-when-built-into-html)
- [ ] If this PR closes an open issue, add the phrase `Closes #ISSUENUMBER` to the description above

*If you are having difficulty fixing build errors, first consult <https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions> carefully, especially ["Common Build Errors"](https://github.com/programminghistorian/jekyll/wiki/Making-Technical-Contributions#common-build-errors). Then contact the technical team if you need further help.*

drjwbaker committed:
Update r-basics-with-tabular-data.md

I think this is wrong. For me l60..

> Take a look at the `AirPassengers` data set by typing `data(AirPassengers)` into the console

..doesn't work. What does work is just typing `AirPassengers`. Can you check this please @anisa-hawes before merging? (as it may just be my setup!)

DanielAlvesLABDH commented:
Olá @JimmyMedeiros82, nós traduzimos esta lição há pouco tempo. Será que temos o mesmo bug ou foi corrigido na tradução? Abraço

anisa-hawes commented:
Thank you @drjwbaker! I will test it today, and will also fix the broken link which is causing the build to fail!

anisa-hawes commented:
Thank you for noticing this error, @drjwbaker! I have tested this on my set up (I work on macOS BigSur v11.5.2) and found that the same was true. The command `data(AirPassengers)` didn't load the data set into my console. But, as you suggested, simply typing `AirPassengers` did!

![Screenshot 2021-12-01 at 11 59 01](https://user-images.githubusercontent.com/87070441/144230675-33acf0ce-c0c5-426f-95d9-156b0df88bc0.png)
 
I can update this on EN, ES and PT.

Also: the link <https://joedicastro.com/pages/markdown.html> actually appears to be live. Perhaps the site was temporarily down when this build check ran...

anisa-hawes commented:
Re-reading this, it seems to me that the whole paragraph is quite confusing. The sentences "Enter `data(AirPassengers)` into the console and hit *Enter*. On the next line, type in `AirPassengers` and hit *Enter* again" are redundant, effectively repeating the same instruction...

My suggestion is that this would be clearer:

Take a look at the AirPassengers data set. To load the data set into your console, type `AirPassengers` and hit Enter. This will bring up a table showing the number of passengers who flew on international airlines between January 1949 and December 1960, in thousands. You should see:

I'll also update the code blocks, so that the commands shown match those instructed in the lesson text.

anisa-hawes committed:
Update r-basics-with-tabular-data.md

Updating para.60, to replace the command `data(AirPassengers)` with ` AirPassengers` and clarify the instruction.
Also, updating the code block, lines 63-4.

anisa-hawes committed:
Update datos-tabulares-en-r.md

Clarifying the instruction para.63.
Updating the code block, lines 66-7.

anisa-hawes committed:
Update nocoes-basicas-R-dados-tabulares.md

Updating para.66, to replace the command `data(AirPassengers)` with `AirPassengers` and clarify the instruction.
Also, updating the code block, lines 69-70.

DanielAlvesLABDH committed:
Update nocoes-basicas-R-dados-tabulares.md

Update the paragraph to eliminate the repetition in Portuguese

rivaquiroga commented:
Hi everyone! 
This is not really a bug. Both commands do different things.
`data(AirPassengers)` makes this built-in dataset available in your Global Environment. Running `AirPassengers` will only print it to the console. Doing the first step is not a requirement for the second one, BUT after loading a dataset with `data()` you will be able to see it in the Environment pane. Sometimes this is useful when teaching, especially if you are using an IDE like RStudio, because students can see the dataset as an object/variable. 

If you only print the object by calling it by its name, your RStudio session will look like this:

![Captura de pantalla de 2021-12-01 09-54-44](https://user-images.githubusercontent.com/31421616/144238731-dbc154d2-842a-44a9-92f3-fad36420ba0a.png)

If you run `data(AirPassengers)`, the dataset is loaded in your environment as a value

![Captura de pantalla de 2021-12-01 09-54-23](https://user-images.githubusercontent.com/31421616/144239082-11b2e31b-7230-4668-af5f-45140543303a.png)

If you print it with `AirPassengers` after that, you will also see information about the dataset in the environment. This will not happen if you skip the `data(AirPassengers)` step.

![Captura de pantalla de 2021-12-01 09-57-31](https://user-images.githubusercontent.com/31421616/144239197-76a64151-861c-47a9-93a9-9037ff2a4a03.png)


Because the lesson is only using the R Console, it looks like `data(AirPassengers)` is not doing anything, but it is. 

This part of the lesson is a little confusing, because that command is not loading the dataset into the console, but into the global environment: _"Take a look at the AirPassengers data set by typing `data(AirPassengers)` into the console. This will load the data set into the console."_. 
If you decide to remove the `data(AirPassengers)` step, I would suggest using the word_print_ and not _load_ to refer to what happens when you just type `AirPassengers`.

drjwbaker commented:
Thanks for your comments @rivaquiroga: shows how much I know about the R console! (not much).

>  If you decide to remove the `data(AirPassengers)` step, I would suggest using the word_print_ and not load to refer to what happens when you just type `AirPassengers`.

That makes the most sense to me as the `data(AirPassengers)` step appears redundant given how the lesson is written.

jenniferisasi commented:
@rivaquiroga, who knows more R than I do, bit me to explain the "issue". Or misunderstanding of the function `data()` (loading) vs printing the dataset. 

Additionally, @anisa-hawes remember in my workshop I said you can Command-click on the dataset and it will open a spreadsheet-like view? This is why :) 

Re-reading the paragraph in my translation, I think it is clear but it's a bit messy so I would be happy to update it and explain it better

anisa-hawes committed:
Update r-basics-with-tabular-data.md

Updating para.60
Updating code block, line 64

anisa-hawes commented:
Thank you @rivaquiroga and @jenniferisasi!  

Taking what you have explained into account, I wonder if the following adjustment I've made to the EN wording might make best sense: 

First, you need to load the AirPassengers data set into your console. Type `data(AirPassengers)` and hit *Enter*[^1]. To view the data set, type in `AirPassengers` on the next line and hit *Enter* again. This will print a table showing the number of passengers who flew on international airlines between January 1949 and December 1960, in thousands. You should see:

> data()
> data(AirPassengers)
> AirPassengers

anisa-hawes commented:
> That makes the most sense to me as the `data(AirPassengers)` step appears redundant given how the lesson is written.

Ah...  Let me know what you think is best, @drjwbaker.

drjwbaker commented:
No no, go with what @rivaquiroga and @jenniferisasi say: they are the experts here, I hardly ever use rconsole. It feels like the article is using best practice for a reason, just hasn't quite explained why.

rivaquiroga commented:
@anisa-hawes I would suggest for the first sentence: "First, you need to load the AirPassengers data set into your R session"

anisa-hawes committed:
Update r-basics-with-tabular-data.md

Adjust first sentence of para.60

anisa-hawes committed:
Update datos-tabulares-en-r.md

Update code block, line 67

rivaquiroga commented on en/lessons/r-basics-with-tabular-data.md:
````diff
@@ -57,9 +57,10 @@ The console is where you will type commands. To clear the initial screen, go to
+First, you need to load the AirPassengers data set into your R session. Type `data(AirPassengers)` and hit *Enter*[^1]. To view the data set, type in `AirPassengers` on the next line and hit *Enter* again. This will print a table showing the number of passengers who flew on international airlines between January 1949 and December 1960, in thousands. You should see:
 
 ```
+> data()
````
Maybe we can remove this extra `data()`, as it is mention in the previous paragraph just as an example of how to take a look to all the available datasets. It is not really a step expected for users to do for what comes next

anisa-hawes replied:
Yes, okay. I'll remove this!

anisa-hawes committed:
Update nocoes-basicas-R-dados-tabulares.md

Update code block, line 70

anisa-hawes committed:
Update r-basics-with-tabular-data.md

Remove `> data()`, line 63

anisa-hawes committed:
Update datos-tabulares-en-r.md

Remove `> data()`, line 66

anisa-hawes committed:
Update nocoes-basicas-R-dados-tabulares.md

Remove `> data()`, line 69

anisa-hawes commented:
Hello @JimmyMedeiros82 and @DanielAlvesLABDH. Might either of you have a moment to review paragraph 60 of nocoes-basicas-R-dados-tabulares.md, to ensure that the instructions are clear about what each command achieves?

`data(AirPassengers)` loads the data set into the R session
`AirPassengers` prints the data set 

The adjusted EN text is:

First, you need to load the AirPassengers data set into your R session. Type `data(AirPassengers)` and hit *Enter*[^1]. To view the data set, type in `AirPassengers` on the next line and hit *Enter* again. This will print a table showing the number of passengers who flew on international airlines between January 1949 and December 1960, in thousands. You should see:

DanielAlvesLABDH committed:
Update nocoes-basicas-R-dados-tabulares.md

Correcting the Portuguese translation

DanielAlvesLABDH committed:
Update nocoes-basicas-R-dados-tabulares.md

Small change in the wording

anisa-hawes commented:
Thank you, @DanielAlvesLABDH!

Let us know if you have time to check the ES for clarity, @jenniferisasi or @rivaquiroga? When you're happy, I'll merge these changes.

jenniferisasi commented:
I'll do it! Give me a sec...

jenniferisasi committed:
Update datos-tabulares-en-r.md

anisa-hawes commented:
Thank you, all!

There's a broken link now, so I will fix this.

anisa-hawes commented:
Hmmm.... link seems to be correct & live....
<https://github.com/impresso/PH-passim-tutorial/blob/master/explore-passim-output.ipynb>
Will re-run...
